cmd/assetsvc/postgresql_utils.go

```diff
@@ -92,11 +101,21 @@ func (m *postgresAssetManager) getPaginatedChartList(namespace, repo string, pag
 	return charts, 1, nil
 }
 
-func (m *postgresAssetManager) getChart(namespace, chartID string) (models.Chart, error) {
+func (m *postgresAssetManager) getChart(namespace, chartIDUnescaped string) (models.Chart, error) {
 	var chart models.ChartIconString
+	chartID, _ := url.PathUnescape(chartIDUnescaped)
 	err := m.QueryOne(&chart, fmt.Sprintf("SELECT info FROM %s WHERE repo_namespace = $1 AND chart_id = $2", dbutils.ChartTable), namespace, chartID)
 	if err != nil {
-		return models.Chart{}, err
```

Comment:
You've put an explanation in the PR description, but that would need to be here also. You said there that: "Added a fallback query when a char/file is not being retrieved. Why? When upgrading a mirrored chart (eg, jfrog/bitnami/wordpress), we only have (at that point) the repo and the name (ie "jfrog" and "Wordpress"). This changes adds a fallback query id="jfrog whatever wordpress" so that it can be retrieved."

What I don't understand is why you say you only have the repo and the name? It appears that you have the repo and the ID (since you're splitting it to get the name). I see it's something to do with a difference when upgrading a mirrored chart - but the `alikeChartID` that you create appears be just transforming "bitnami/wordpress" to "bitnami%wordpress" ? What is it actually matching in this case in the DB?

Reply:
>This changes adds a fallback query id="jfrog whatever wordpress" so that it can be retrieved

If that's correct, we may be retrieving the wrong wordpress (i.e. jfrog/impersonator/wordpress).

Reply:
> > This changes adds a fallback query id="jfrog whatever wordpress" so that it can be retrieved
> 
> If that's correct, we may be retrieving the wrong wordpress (i.e. jfrog/impersonator/wordpress).

Yes, you're right. Indeed, I wanted to discuss this point with you: 
AFAIK, from Helm we're not able to determine whether the release of the chart `bitnami/wordpress` comes from `bitnami/wordpress`, `jfrog/bitnami/wordpress` or even `jfrog/impersonator/wordpress`.

Given that I just assumed _ok, as a chart repository, I don't want to have repeated charts, so, if I mirror something, I'd ensure there is solely one chart_. It is a too optimistic assumption, I know.
With this solution/workaround: it definitively may retrieve the wrong chart, but, I wasn't able to come up with a better solution. 
Note that, without this, users won't be able to upgrade charts from repos with slashes. That's why I opted for this solution.

If you have an alternative approach to overcome this problem, I really will appreciate it! If not, perhaps we should change the frontend logic and let the user decide which chart to choose if the fallback query returns more than one result.

Reply:
_talked about this in slack_

Let's do two different calls, one expecting exact matches (for the deployment use case) and one for a non-exact match (for the upgrade use case, in which we only have the repo id and the chart name).

Also, not sue if it will be included in this PR, it would be ideal to return the list of repos that contain the chart if more than one matches the query.